README.md

```diff
@@ -37,6 +37,36 @@ The `azure.blob` plugin stores data in Azure Blob Storage.
 1. [Copy the connection string][connstring] for the storage account. Then set it as an environment variable named 
     `AZURE_STORAGE_CONNECTION_STRING`.
 
+## Secrets
+
+Secrets plugins allow Porter to inject secrets into credential sets.
+
+For example, your team may have a shared key vault with a database password, you can use the keyvault plugin
+to use it as a credential when you install a bundle.
+
+### Key Vault
+
+The `azure.keyvault` plugin resolves credentials against secrets in Azure Key Vault.
+
+1. Open, or create, `~/.porter/config.toml`
+1. Add the following lines to activate the Azure keyvault secrets plugin:
+
+    ```toml
+    default-secrets = "mysecrets"
+    
+    [[secrets]]
+    name = "mysecrets"
+    plugin = "azure.keyvault"
+    
+    [secrets.config]
+    vault = "myvault"
+    ```
+1. [Create a key vault][keyvault] and set the vault name in the config with name of the vault.
+1. [Create a service principal][sp] and create an Access Policy on the vault giving Get and List secret permissions.
+1. Using credentials for the service principal set the environment variables: `AZURE_TENANT_ID`,`AZURE_CLIENT_ID`,  and `AZURE_CLIENT_SECRET`.
```

Comment:
Do we want to add docs (here or elsewhere) around the use case of setting/using a service principal prefix?

Reply:
I didn't want to document beyond the one case that I know we want to support. I'm still playing around with how to best deal with an auth solution that requires the least effort for users. So as part of that follow-up I mentioned, I'll document whatever we settle on.